ngraph/core/reference/include/ngraph/runtime/reference/transpose.hpp

```diff
@@ -20,40 +18,69 @@ namespace ngraph
     {
         namespace reference
         {
-            template <typename T, typename U>
-            void transpose(const T* arg, T* out, Shape arg_size, const U* axes_order = nullptr)
+            void transpose(const char* data,
+                           char* out,
+                           const Shape& data_shape,
+                           size_t element_size,
+                           const int64_t* axes_order = nullptr,
```

Comment:
Looks like in the transpose operation we always use this function without default parameters.
Maybe in this case we can remove default values for parameters.

Moreover `axes_order` also is not equal to `nullptr`. We can get the `nullptr` only from legacy implementation, which is used by ARM plugin. I propose to move logic for cases `axes_order == nullptr` and `out_shape.empty()` to this method.

Reply:
You are right, I wanted to preserve some Transpose output calculation logic in the reference, instead of just calling opt_kernel::reshape implementation. 
As you suggested, I moved the default params behaviour to the legacy function, so the new reference always reuses output shape calculated by evaluate method. Please take a look (commit https://github.com/openvinotoolkit/openvino/pull/5315/commits/c38f0491cfa7d9d712793c6ebcd5978f429a5919)